core/storage/suggestion/gae_models.py

```diff
@@ -386,3 +389,82 @@ def update_position_in_rotation(cls, score_category, user_id):
         else:
             instance.current_position_in_rotation = user_id
             instance.put()
+
+
+class GeneralVoiceoverApplicationModel(base_models.BaseModel):
+    """A general model for voiceover application of an entity.
+
+    The ID of the voiceover application will be a random hashed value.
+    """
+    # The type of entity to which the user will be assigned as a voice artist
+    # once the application will get approved.
+    target_type = ndb.StringProperty(required=True, indexed=True)
+    # The ID of the entity to which the application belongs.
+    target_id = ndb.StringProperty(required=True, indexed=True)
+    # The language code for the voiceover audio.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+    # The status of the application. One of: accepted, rejected, in-review.
+    status = ndb.StringProperty(
+        required=True, indexed=True, choices=STATUS_CHOICES)
+    # The HTML content written in the given language_code.
+    # This will typically be a snapshot of the content of the initial card of
+    # the target.
+    content = ndb.StringProperty(required=True, indexed=True)
+    # The filename of the voiceover audio. The filename will have
+    # datetime-randomId(length 6)-language_code.mp3 pattern.
+    filename = ndb.StringProperty(required=True, indexed=True)
+    # The ID of the author of the voiceover application.
+    author_id = ndb.StringProperty(required=True, indexed=True)
+    # The ID of the reviewer who accepted/rejected the voiceover application.
+    final_reviewer_id = ndb.StringProperty(indexed=True)
+    # The plain text message submitted by the reviewer while rejecting the
+    # application.
+    rejection_message = ndb.StringProperty(indexed=True)
+
+    @staticmethod
+    def get_deletion_policy():
+        """General voiceover application needs to be pseudonymized for the
+        user.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether GeneralVoiceoverApplicationModel exists for the user.
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            ndb.OR(cls.author_id == user_id, cls.final_reviewer_id == user_id)
+        ).get() is not None
+
+    @classmethod
+    def get_user_voiceover_applications(cls, author_id, status=None):
```

Comment:
Do we really need the default parameter?

Reply:
This default parameter will help us to fetch data based on the `status` value if it's None we aren't going to fetch data based on the status value.
Usecase:
- To view the accepted application we will pass status=accepted.
- To view, all user submitted an application which is in the review.
- To fetch all the applications submitted by the user. (This will be used in services to make sure that user is allowed to submit a new application.)

Reply:
I'll update the docstring!

Reply:
Maybe we can pick a better non-null value for getting all user-submitted applications in review? I think as far as possible we should try and avoid default parameters

Reply:
> I think as far as possible we should try and avoid default parameters

I agree with this but I'm not sure what I can do here, do you have any suggestions?

Reply:
Default value of None in this case is OK -- that's a reasonably common paradigm in situations like this where there are optional params added to a query.

But this function docstring could probably use more documentation about the possible values of "status" and their effects on the query.